src/components/Select.jsx

```diff
@@ -35,16 +59,27 @@ const Select = ({
           </div>
         )}
         {editable && (
-          <Dropdown.Menu className="w-full bg-hackathon-green-100 !border-none !rounded-none !p-0 overflow-y-auto max-h-[35vh]">
-            {options.map((option, index) => (
-              <Dropdown.Item
-                className=" hover:!bg-hackathon-green-200 !bg-hackathon-green-100 overflow-hidden"
-                key={index}
-                onClick={() => setUser({ ...user, [field]: option })}
-              >
-                {option}
-              </Dropdown.Item>
-            ))}
+          <Dropdown.Menu className="w-full !bg-hackathon-green-100 !border-none !rounded-none !p-0 overflow-y-auto max-h-[35vh]">
+            {searchable && (
+              <input
+                autoFocus
+                className="mx-1.5 my-1 w-11/12 ring-0 outline-none px-2 py-1 bg-hackathon-green-100"
+                placeholder="Search"
+                onChange={handleInput}
+              />
+            )}
+            {options.map(
+              (option, index) =>
+                !option.hidden && (
```

Comment:
if ur only showing visible elements, then u can filter and then map, it will be faster